src/object_store/src/object/opendal_engine/opendal_s3.rs

```diff
@@ -78,4 +79,56 @@ impl OpendalObjectStore {
 
         Ok(HttpClient::build(client_builder)?)
     }
+
+    /// currently used by snowflake sink,
+    /// especially when sinking to the intermediate s3 bucket.
+    pub fn new_s3_engine_with_credentials(
```

Comment:
Since our current streaming uploader of opendal s3 object store is just a simpler wrapper around the raw opendal s3 writer, I think we can just use the raw opendal s3 writer directly in the sink writer instead of reusing the streaming uploader of the object store crate. 

Besides, the newly added method is only for snowflake sink, but it's wired to read the env var `RW_S3_ENDPOINT` and `RW_IS_FORCE_PATH_STYLE`, which should only be used by our storage engine.

Reply:
> Since our current streaming uploader of opendal s3 object store is just a simpler wrapper around the raw opendal s3 writer, I think we can just use the raw opendal s3 writer directly in the sink writer instead of reusing the streaming uploader of the object store crate.

actually I'm in favor of wrapping things around - since we essentially need to provide those nitty-gritty stuff (e.g., s3 credentials, op configuration, etc.), thus the existing unified wrapper struct looks good to me.

plus, I remove the useless env var loading from the `new_s3_engine_with_credentials`.